lua/hbac/telescope/init.lua

```diff
@@ -16,20 +16,31 @@ local telescope_conf = require("telescope.config").values
 
 local M = {}
 
+local attach_mappings = function(opts)
+	return function(_, map)
+		local telescope_mappings = require("hbac.config").values.telescope.mappings
+		telescope_mappings = vim.tbl_deep_extend("force", telescope_mappings, opts.mappings or {})
+		for mode, hbac_telescope_mappings in pairs(telescope_mappings) do
+			for key, action in pairs(hbac_telescope_mappings) do
+				map(mode, key, action)
+			end
+		end
+		return true
+	end
+end
+
 M.pin_picker = function(opts)
-	local attach_mappings = require("hbac.telescope.attach_mappings")
 	local make_finder = require("hbac.telescope.make_finder")
 	opts = opts or {}
-	pickers
-		.new(opts, {
-			prompt_title = "Hbac Pin States",
-			finder = make_finder.make_finder(opts),
-			sorter = telescope_conf.generic_sorter(opts),
-			attach_mappings = attach_mappings.attach_mappings,
-			previewer = telescope_conf.file_previewer(opts),
-			default_selection_index = make_finder.default_selection_idx,
-		})
-		:find()
+	pickers.new(opts, {
```

Comment:
Alternatively to the other suggestion
```suggestion
	pickers
	    .new(opts, {
```